src/main/java/data/SubjectRawDataBuilder.java

```diff
@@ -0,0 +1,80 @@
+package data;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Date;
+import java.util.Scanner;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * Created by Timotej on 21-Oct-15.
+ */
+public class SubjectRawDataBuilder {
+
+    private Integer id;
+    private File closeMic;
+    private File BPMData;
+
+    public SubjectRawDataBuilder withId(int id) {
+        this.id = id;
+        return this;
+    }
+
+    public SubjectRawDataBuilder withCloseMic(File closeMic) {
+        this.closeMic = closeMic;
+        return this;
+    }
+
+    public SubjectRawDataBuilder withBPMData(File BPMData) {
+        this.BPMData = BPMData;
+        return this;
+    }
+
+    /**
+     * This methods finds closeMic file and BPMData files from the subject folder
+     * @param subjectDir
+     * @return
+     */
+    public SubjectRawDataBuilder asBCMDir(File subjectDir) {
+        for(File f : subjectDir.listFiles()) {
+            System.out.println(f.getName());
+            if(f.getName().equals("BPM.txt")) {
+                BPMData = f;
+            }
+            else if(f.getName().endsWith("close.wav")) {
+                closeMic = f;
+            }
+        }
+
+        return this;
+    }
```

Comment:
It is debbuging, removed it. There can only be one BPM.txt I think in the folder so that's fine. close.wav can match more than 1 things, but for all the subjects in MBC database I saw, it can match only one. I think this is good enough at the moment. I'll rename it to fromMBCDir, MBC is the shortest name for the current dataset that I saw ...